### PR TITLE
DOCS: Update OSSL_DECODER_CTX_new_by_EVP_PKEY.pod to match declarations

### LINKED ISSUE #13441: Documentation for OSSL_DECODER_CTX_new_by_EVP_PKEY is out of date
The parameters to  `OSSL_DECODER_CTX_new_by_EVP_PKEY` have changed but the documentation has not kept up.

The headers think the function looks like this:

````
OSSL_DECODER_CTX *
OSSL_DECODER_CTX_new_by_EVP_PKEY(EVP_PKEY **pkey,
                                 const char *input_type,
                                 const char *input_struct,
                                 const char *keytype, int selection,
                                 OSSL_LIB_CTX *libctx, const char *propquery);
````

But the documentation thinks the function looks like this:

````
OSSL_DECODER_CTX *
 OSSL_DECODER_CTX_new_by_EVP_PKEY(const EVP_PKEY *pkey,
                                  const char *input_type, const char *keytype,
                                  OSSL_LIB_CTX *libctx, const char *propquery);
````
Note: missing `input_struct` and `selection` parameters.

### PR DESCRIPTION
**Pull request by levitte**
Fixes #13441

### CODE REVIEW

File: doc/man3/OSSL_DECODER_CTX_new_by_EVP_PKEY.pod [hunk 1]
```diff
@@ -36,23 +38,27 @@ OSSL_DECODER_CTX_set_passphrase_cb
...
-The search of decoder implementations can also be limited with I<keytype>,
-which specifies the expected resulting keytype.  NULL is valid input and
-signifies that the decoder implementations will find out the keytype on
-their own from the input they get.
+The search of decoder implementations can be limited with I<input_type> and
+I<input_struct> which specifies a starting input type and input structure.
+NULL is valid for both of them and signifies that the decoder implementations
+will find out the input type on their own.
+This is further explained in L<OSSL_DECODER_CTX_set_input_type(3)>
+and L<OSSL_DECODER_CTX_set_input_structure(3)>.
```
Comment (mattcaswell) [inline doc/man3/OSSL_DECODER_CTX_new_by_EVP_PKEY.pod:56]: How can the reader find out what valid values are for the input structure? The above link doesn't give any clue. Similarly for the input_type.

Comment (levitte) [reply doc/man3/OSSL_DECODER_CTX_new_by_EVP_PKEY.pod]: I took a stab at this, and included the start of openssl-glossary(7)

**Comment by levitte** [comment]
CIs are happy

File: doc/man3/OSSL_DECODER_CTX_new_by_EVP_PKEY.pod [hunk 2]
```diff
@@ -36,23 +38,28 @@ OSSL_DECODER_CTX_set_passphrase_cb
 OSSL_DECODER_CTX_new_by_EVP_PKEY() is a utility function that creates a
 B<OSSL_DECODER_CTX>, finds all applicable decoder implementations and sets
 them up, so all the caller has to do next is call functions like
-OSSL_DECODE_from_bio().
+L<OSSL_DECODER_from_bio(3)>.  The caller may used the optional I<input_type>,
```
Comment (t8m) [inline doc/man3/OSSL_DECODER_CTX_new_by_EVP_PKEY.pod:41]: s/may used/may use/ ?

File: doc/man3/OSSL_DECODER_CTX_new_by_EVP_PKEY.pod [hunk 3]
```diff
@@ -36,23 +38,28 @@ OSSL_DECODER_CTX_set_passphrase_cb
...
+L<OSSL_DECODER_CTX_set_input_structure(3)>.
+See L</Input Types> and L</Input Structures> below for further information.
 
-The search of decoder implementations can also be limited with I<keytype>,
-which specifies the expected resulting keytype.  NULL is valid input and
-signifies that the decoder implementations will find out the keytype on
-their own from the input they get.
+The search of decoder implementations can also be limited with I<keytype>
+and I<selection>, which specifies the expected resulting keytype and contents.
+NULL and zero are valid and signifies that the decoder implementations will
```
Comment (t8m) [inline doc/man3/OSSL_DECODER_CTX_new_by_EVP_PKEY.pod:61]: s/signifies/signify/

File: doc/man7/openssl-glossary.pod [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+L<https://www.itu.int/rec/T-REC-X.690>
+
+=item MSBLOB
+
+MSBLOB is a Microsoft specific binary format for RSA and DSA keys, both
+private and public.  This form is never passphrase protected.
+
+=item PEM ("Privacy Enhanced Message")
+
+PEM is a format used for encoding of binary content into an mail and ASCII
```
Comment (t8m) [inline doc/man7/openssl-glossary.pod:41]: s/an mail/a mail/

File: doc/man7/openssl-glossary.pod [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+friendly form.  The content is a series of base64-encoded lines, surrounded
+by begin/end markers each on their own line.  For example:
+
+ -----BEGIN PRIVATE KEY-----
+ MIICdg....
+ ... bhTQ==
+ -----END PRIVATE KEY-----
+
+Optional header line(s) may appear after the begin line, and their existence
+depends on the type of object being written or read.
```
Comment (t8m) [inline doc/man7/openssl-glossary.pod:51]: the object?

File: doc/man7/openssl-glossary.pod [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+For all OpenSSL uses, the binary content is expected to be a DER encoded
+structure.
+
+This is defined in IETF RFC 1421:
+
+L<https://tools.ietf.org/html/rfc1421>
+
+=item PKCS#8 (also known as "pkcs8" in some parts of OpenSSL)
+
+PKCS#8 is a specification an ASN.1 structure that OpenSSL uses for
```
Comment (t8m) [inline doc/man7/openssl-glossary.pod:62]: of an ASN.1

File: doc/man7/openssl-glossary.pod [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+storing or transmitting any private key in a key type agnostic
+manner, and has both an unencrypted and an encrypted form.
+
+This is specified in RFC 5208:
+
+L<https://tools.ietf.org/html/rfc5208>
+
+=item PVK
+
+MSBLOB is a Microsoft specific binary format for RSA and DSA private
```
Comment (t8m) [inline doc/man7/openssl-glossary.pod:72]: s/MSBLOB/PVK/

**Review by t8m — CHANGES_REQUESTED** [review]
Just a few typos/nits.

File: doc/man7/openssl-glossary.pod (hunk 5 above)
Comment (levitte) [reply doc/man7/openssl-glossary.pod:51]: Yeah, this is the original string that I copied verbatim.  I don't have any better idea...

File: doc/man3/OSSL_DECODER_CTX_new_by_EVP_PKEY.pod (hunk 2 above)
Comment (levitte) [reply doc/man3/OSSL_DECODER_CTX_new_by_EVP_PKEY.pod:41]: Yup.  Fixed

File: doc/man3/OSSL_DECODER_CTX_new_by_EVP_PKEY.pod (hunk 3 above)
Comment (levitte) [reply doc/man3/OSSL_DECODER_CTX_new_by_EVP_PKEY.pod:61]: Fixed

File: doc/man7/openssl-glossary.pod (hunk 4 above)
Comment (levitte) [reply doc/man7/openssl-glossary.pod:41]: Fixed

File: doc/man7/openssl-glossary.pod (hunk 6 above)
Comment (levitte) [reply doc/man7/openssl-glossary.pod:62]: Fixed

File: doc/man7/openssl-glossary.pod (hunk 7 above)
Comment (levitte) [reply doc/man7/openssl-glossary.pod:72]: Thanks.  Fixed

**Review by t8m — APPROVED** [review]

**Comment by levitte** [comment]
Merged

19ad83f6c8856b6c28087ff96456b17c471b299a DOCS: Update OSSL_DECODER_CTX_new_by_EVP_PKEY.pod to match declarations

**Comment by levitte** [comment]
Let's help each other to remember doc/man7/openssl-glossary.pod, yeah?

**Pull request closed without merging**